tests/core/caching-utils/test_request_caching.py

```diff
@@ -260,30 +265,31 @@ def test_blocknum_validation_against_validation_threshold_when_caching(
         ("pending", None, False),
     ),
 )
-def test_block_id_param_caching(
+def test_block_id_param_caching_mainnet(
     threshold, endpoint, block_id, blocknum, should_cache, request_mocker
 ):
     w3 = Web3(
-        HTTPProvider(
+        BaseProvider(
             cache_allowed_requests=True, request_cache_validation_threshold=threshold
         )
     )
     with request_mocker(
         w3,
         mock_results={
+            "eth_chainId": "0x1",  # mainnet
             endpoint: "0x0",
             "eth_getBlockByNumber": lambda _method, params: (
                 # mock the threshold block to be blocknum "0x2" for all test cases
-                {"number": "0x2"}
+                {"number": "0x2", "timestamp": "0x0"}
                 if params[0] == threshold.value
-                else {"number": blocknum}
+                else {"number": blocknum, "timestamp": "0x0"}
             ),
         },
     ):
         assert len(w3.provider._request_cache.items()) == 0
         w3.manager.request_blocking(RPCEndpoint(endpoint), [block_id, False])
         cached_items = len(w3.provider._request_cache.items())
-        assert cached_items == 1 if should_cache else cached_items == 0
+        assert cached_items > 0 if should_cache else cached_items == 0
```

Comment:
oops, that was me playing around with some different ideas I think. Thought I had changed them all back. This was a result of not turning off caching when making requests to get validation blocks, etc. But it's not a good approach since it can enter recursion. I'll change this back, good catch 👍🏼